frontend/src/cljs/imas_seamap/state_of_knowledge/events.cljs

```diff
@@ -369,3 +369,27 @@
         db     (assoc-in db [:map :bounds] bounds)]
     {:db       db
      :put-hash (encode-state db)}))
+
+(defn habitat-toggle-show-layers [{:keys [db]} _]
+  (let [db             (update-in db [:state-of-knowledge :statistics :habitat :show-layers?] not)
+        show-layers?   (get-in db [:state-of-knowledge :statistics :habitat :show-layers?])
+        habitat-layers (get-in db [:map :keyed-layers :habitat])]
+    {:db db
+     :dispatch-n
+     (mapv #(vector (if show-layers? :map/add-layer :map/remove-layer) %) habitat-layers)}))
+
+(defn bathymetry-toggle-show-layers [{:keys [db]} _]
+  (let [db               (update-in db [:state-of-knowledge :statistics :bathymetry :show-layers?] not)
+        show-layers?     (get-in db [:state-of-knowledge :statistics :bathymetry :show-layers?])
+        bathymetry-layers (get-in db [:map :keyed-layers :bathymetry])]
+    {:db db
+     :dispatch-n
+     (mapv #(vector (if show-layers? :map/add-layer :map/remove-layer) %) bathymetry-layers)}))
+
```

Comment:
Just musing, but this kind of duplication makes me wonder about more efficient formats.  I think we could review subs and events in particular at some point.  (possibly also, the whole events vs actions or whatever Oli's distinction was; there's a lot of raw app-db manipulatin happening that could probably be made tidier behind utils for eg)